app/src/main/java/org/mozilla/fenix/perf/LazyMonitored.kt

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.perf
+
+import mozilla.components.support.base.log.logger.Logger
+import java.util.concurrent.atomic.AtomicInteger
+
+private val logger = Logger("LazyMonitored")
+
+/**
+ * A container for the number of components initialized.
+ */
+object ComponentInitCount {
+    val count = AtomicInteger(0)
+}
+
+/**
+ * A convenience function for setting the [LazyMonitored] property delegate, which wraps
+ * [lazy] to add performance monitoring.
+ */
+fun <T> lazyMonitored(initializer: () -> T) = LazyMonitored(initializer)
+
+/**
+ * A wrapper around the [lazy] property delegate to monitor for performance related issues.
+ * For example, we can count the number of components initialized to see how the number of
+ * components initialized on start up impacts start up time.
+ */
+class LazyMonitored<T>(initializer: () -> T) {
```

Comment:
Could we put this private, and just say `lazyMonitored` returns a `Lazy<T>` type? or is this not feasible?

Reply:
Done. It wasn't possible before because you can't expose a private class through a non-private function but now I can hide it behind `Lazy` instead.